File: arrays-and-strings/1.4.js
function isPalindromePermutation(str) {
  var charCounts = {};
  str = str.toLowerCase();
  for (var i = 0; i < str.length; i++) {
    var char = str.charAt(i);
    if (char.match(/[a-z]/g)) {
      charCounts[char] = charCounts[char] ? charCounts[char] + 1 : 1;
    }
  }

  return !hasMoreThanOneOdd(charCounts);
}

function hasMoreThanOneOdd(counts) {
  var foundOdd = false;
  for (var key in counts) {
    var count = counts[key]
    if (count % 2 !== 0) {
      if (foundOdd) return true;
      foundOdd = true;
    }
  }

  return false;
}

File: arrays-and-strings/1.4.py
import re

def is_palindrome_permutation(string):
    char_counts = {}
    string = string.lower()
    for char in string:
        if re.match(r'[a-z]', char):
            char_counts[char] = 1 if not char_counts.get(char) else char_counts[char] + 1
    return not has_more_than_one_odd(char_counts)

def has_more_than_one_odd(counts):
    found_odd = False
    for key in counts:
        count = counts[key]
        if count % 2 != 0:
            if found_odd:
                return True
            found_odd = True
    return False
